src/sagemaker/model_card/model_card.py

```diff
@@ -898,8 +1152,13 @@ def __init__(
         self.training_details = training_details
         self.evaluation_details = evaluation_details
         self.additional_information = additional_information
+        self.model_package_details = self._from_model_package(model_package)
         self.sagemaker_session = sagemaker_session or Session()
 
+    def _from_model_package(self, model_package: ModelPackage):
```

Comment:
Is the intent to left the implementation for another PR?  If so, I think we should remove this function from this PR and add it later since it is not necessarily aligned with the purpose of this PR, auto-discover model package metadata so that we could keep this PR clean

Reply:
Yes, this is for another PR specially for Xiaotong's work. She can pull these changes after getting merged and continue working from this function.

Reply:
OK. If that is the case, I would suggest to let Xiaotong to add function `_from_model_package`. As a good practice, I think PR should only include what it is intended.